### DISCUSSION TITLE
How do I include a template in a library crate without getting spurious errors from Cargo when building the generated project?

### DISCUSSION THREAD

**Discussion by petekubiak** [cargo-generate/cargo-generate#1353] [Q&A] [answered]
I've created a library crate and want to include a template for a binary crate which imports the library and demonstrates a simple example of its use. I've managed to do this and everything works, however when building the generated project I get an error from cargo like the following:
```text
error: invalid character `{` in package name: `{{project-name}}`, the first character must be a Unicode XID start character (most letters or `_`)
 --> ../../.cargo/git/checkouts/commitment-issues-3d6fdbd244e07b67/159b725/template/Cargo.toml:2:8
  |
2 | name = "{{project-name}}"
  |        ^^^^^^^^^^^^^^^^^^
  |
```
It appears that cargo is picking up on the Cargo.toml in the template and throwing this error, even though the template isn't used in the library. The build succeeds, but I would like to remove the error if possible.

Has anyone else encountered this? Does anyone have a suggestion for a workaround? I've tried using [package.exclude] and [package.include], but I think those only work for crates published on crates.io, not ones pulled from git

**Comment by petekubiak** [comment]
Ok I've come up with a solution, but I'm keen to know if there's a better one.

I renamed the template's `Cargo.toml` -> `Cargo.toml.template`.

Then I added this to `cargo-generate.toml`:
```toml
[hooks]
post = ["move-cargo-toml.rhai"]
```

and finally created `move-cargo-toml.rhai`:
```rhai
file::rename("Cargo.toml.template", "Cargo.toml")
```

**Comment by sassman** [comment, marked as answer]
The most easiest is to name the files `.liquid`. This extension is then automatically removed. 

Also you could have a `Cargo.toml` with your resulting project (for demo or testing purpose) and a `Cargo.toml.liquid`, when the template is rendered the `Cargo.toml` will be overwritten by the `Cargo.toml.liquid` rendered version (and named `Cargo.toml`)
